### PR TITLE
feat(functions): #1556 saved historic builds

### LINKED ISSUE #1556: Save historic build time
Save the build time history each time a new time is updated in sub collection `repositories/uid/builds`

- time
- PR / commit information (so we can create a link on the UI)
- Webhook to save the latest data (append existing data)
- when deleting the repository, we must also delete the sub collection `builds` first

### PR DESCRIPTION
**Pull request by webkhushboo**
closes #1556 

### Notes

A summary of what was achieved in this PR

- [ ] API: Save pull request statuses data 
- [ ] API: Delete statuses collection when project delete
- [ ] API: Delete statuses collection when repository update
- [ ] API: When webhook calls it update the historic build time

### CODE REVIEW

File: functions/src/project/delete-project.ts [hunk 1]
```diff
@@ -29,6 +30,8 @@ async function deleteProjectRepositories(projectUid: string, project: ProjectMod
               }
 
               t.update(repoRef, { projects: repoData.projects });
+            
+              deleteRepoBuilds(repositoryUid);
```
Comment (eddiejaoude) [inline functions/src/project/delete-project.ts]: this is a promise and should be handled. I fixed this, may be it was in a commit in the old PR

Comment (webkhushboo) [reply functions/src/project/delete-project.ts]: This has been commented in the https://github.com/DashboardHub/PipelineDashboard/pull/1604/files
I don't know why its visible here.

**Comment by eddiejaoude** [comment]
Can you add a screenshot of the data saved in the DB?

I seem to have missing data in the tests I did...

## CI builds

Screenshot 2019-12-29 21 47 53

## Statuses is an empty array

Screenshot 2019-12-29 21 47 22

**Comment by webkhushboo** [comment]
@eddiejaoude 

Please see the below screenshot with data.

image

**Comment by eddiejaoude** [comment]
It works for Circle CI and Travis CI but not GitHub actions. We can look into GitHub actions in a new PR.

See data from Travis CI

Screenshot 2020-01-01 21 33 43

However, when the PR is closed the `statuses` data remains, this sub collection will get very, so the  document in the sub collection for that PR should get deleted when the webhook for closing that PR comes in (sorry this was not included in the original information)

**Comment by webkhushboo** [comment]
@eddiejaoude 
We will be showing the historic data for build statuses in graphs so we need this data and we can't delete it.

**Comment by eddiejaoude** [comment]
Yes you are right 👍

**Review by eddiejaoude — APPROVED** [review]

**Pull request merged**